### PR TITLE
[iOS] - Add ability to set custom command-line switches

### PR DESCRIPTION
**Pull request by Brandon-T**
## Summary
- Add ability to set custom command-line switches/flags for Brave-Core

Resolves https://github.com/brave/brave-browser/issues/39384

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Enable the Debug menu in Settings
2. Scroll down to Brave Core Switches
3. Scroll down to "Add Custom Switch"
4. Enter Key: "env-ai-chat.bsg" and Value: "dev"
5. Save
6. Notice new switch added at the bottom of the screen with `--env-ai-chat.bsg=dev` as the sub-text.
7. Restart the app
8. Visit: `brave://version` and verify the command-line switch was added.
9. Swipe to delete the switch you added.
10. Verify that it was deleted in the UI.
11. Visit: `brave://version` and verify the command-line switch was removed.

## Screenshots

image
image
image
image

### CODE REVIEW

**Review by stoletheminerals — APPROVED** [review]

File: ios/brave-ios/Sources/Brave/Frontend/Settings/Debug/BraveCoreDebugSwitchesView.swift [hunk 1]
```diff
@@ -173,6 +173,68 @@ private struct BasicPickerInputView: View {
   }
 }
 
+private struct CustomSwitchInputView: View {
+  @ObservedObject private var customSwitches = Preferences.BraveCore.customSwitches
+  @ObservedObject private var switchValues = Preferences.BraveCore.switchValues
+  @Environment(\.presentationMode) @Binding private var presentationMode
```
Comment (kylehickinson) [inline ios/brave-ios/Sources/Brave/Frontend/Settings/Debug/BraveCoreDebugSwitchesView.swift:179]: Prefer `@Environment(\.dismiss) private var dismiss` instead

File: ios/brave-ios/Sources/Brave/Frontend/Settings/Debug/BraveCoreDebugSwitchesView.swift [hunk 2]
```diff
@@ -173,6 +173,68 @@ private struct BasicPickerInputView: View {
...
+
+  @State var key: String = ""
```
Comment (kylehickinson) [inline ios/brave-ios/Sources/Brave/Frontend/Settings/Debug/BraveCoreDebugSwitchesView.swift:181]: Probably shouldn't initialize `State` like this, best to pass in a regular `var` and `onAppear` set the State with that. If you're fine with the `key` passed in being ignored on subsequent body evaluations you can also make a explicit init method that sets the state like `_key = State(initialValue: key)` instead

File: ios/brave-ios/Sources/Brave/Frontend/Settings/Debug/BraveCoreDebugSwitchesView.swift [hunk 3]
```diff
@@ -339,6 +401,56 @@ struct BraveCoreDebugSwitchesView: View {
...
+                      .lineLimit(1)
+                  }
+                }
+              }
+            }
+            .onDelete {
+              let keys = $0.map { Preferences.BraveCore.customSwitches.value[$0] }
+
+              withAnimation(.default) {
+                keys.forEach { key in
```
Comment (kylehickinson) [inline ios/brave-ios/Sources/Brave/Frontend/Settings/Debug/BraveCoreDebugSwitchesView.swift:432]: Prefer regular for loops

**Review by kylehickinson — APPROVED** [review]

**Comment by stephendonner** [comment]
## Verification `IN-PROGRESS` using

```
Version `1.69 (57)`
```

**Pull request merged**